### DISCUSSION TITLE
Closing tabs behaviour

### DISCUSSION THREAD

**Discussion by SlugCamargo** [Able-Bear-Software/tab-nodes-tree#31] [Bug Report]
I'm having a somewhat unstable behaviour when it comes to activating tabs after closing. I'm pretty sure it's a bug. 

I have the following setup:

![image](https://github.com/user-attachments/assets/4bf1eaf0-e9d0-4635-822f-dc2ef85a5e26)

My intention is that upon closing tab 4, tab 3 would be activated. Similarly, upon closing tab 3, tab 2 should be activated; and upon closing tab C, tab B should be activated. 

Instead, about nine times out of ten, closing tabs 2, 3 or 4 will take me to tab 1; whereas closing tab B or C will take me to tab A. It doesn't matter what I do with the extension settings, the behaviour is *almost* always the same - and that's the thing: once in a while I do get the expected behaviour, but I have no clue why.

This is my current setup:

![image](https://github.com/user-attachments/assets/2bd0b814-34c4-422c-9166-3b81a1fb5181)

It doesn't matter whether I'm nesting tabs within tabs, or if I'm using branches; the same thing happens. I've been trying stuff around and, from what I gather, it would appear as though the option _"When the closed tab node has a parent tab node, Activate the parent tab node."_ has priority over everything else, no matter where I put it in the list (and I can't disable it either).

Any help would be greatly appreciated.

**Comment by zeroarst** [comment]
@SlugCamargo Thank you for reporting this issue. I can confirm this is a bug. 
https://github.com/Able-Bear-Software/tab-nodes-tree/issues/32

**Comment by zeroarst** [comment]
Fixed in [v1.4.0](https://github.com/Able-Bear-Software/tab-nodes-tree/discussions/40).
